### PR TITLE
Ask CFPB: Hide redirected pages from Category pages

### PR DESCRIPTION
**Pull request by higs4281**
Redirected pages in Ask CFPB need to remain live in order to do their
job of redirecting requests to another page. But Category and
SubCategory pages were not excluding redirected pages from their
collection of associated pages, resulting in some duplication in the
list of answers offered.

Addresses GHE issue 1292.

### CODE REVIEW

File: cfgov/ask_cfpb/tests/models/test_pages.py [hunk 1]
```diff
@@ -828,21 +828,19 @@ def test_category_page_context_es(self):
...
     def test_category_page_context_no_es(self, mock_es_query):
+        blank_facets = {'answers': {},
+                        'audiences': {},
+                        'subcategories': {}}
         mock_return_value = mock.Mock()
-        mock_return_value.facet_map = json.dumps(
-            {'answers': {},
-             'audiences': {},
-             'subcategories': {}})
+        mock_return_value.facet_map = json.dumps(blank_facets)
```
Comment (hillaryj) [inline cfgov/ask_cfpb/tests/models/test_pages.py]: Should `facet_map` here be `facets`?

File: cfgov/ask_cfpb/models/pages.py [hunk 2]
```diff
@@ -249,7 +249,7 @@ def get_context(self, request, *args, **kwargs):
         context.update({
             'answers': answers,
             'audiences': audiences,
-            'facet_map': facet_map,
+            'facets': facet_dict,
```
Comment (hillaryj) [inline cfgov/ask_cfpb/models/pages.py]: Is this change to distinguish it from the `facet_map()` function?

File: cfgov/ask_cfpb/tests/models/test_pages.py [hunk 3]
```diff
@@ -828,21 +828,19 @@ def test_category_page_context_es(self):
...
         mock_site = mock.Mock()
         mock_site.hostname = 'localhost'
         mock_request = HttpRequest()
         mock_request.site = mock_site
         cat_page = self.create_category_page(ask_category=self.category)
         test_context = cat_page.get_context(mock_request)
-        self.assertEqual(
-            test_context['facet_map'],
-            mock_return_value.facet_map)
+        self.assertEqual(test_context['facets'], blank_facets)
```
Comment (hillaryj) [inline cfgov/ask_cfpb/tests/models/test_pages.py]: Is there a test already for checking that redirects don't make it in the list? If not, let's add one!

File: cfgov/ask_cfpb/tests/models/test_pages.py (hunk 1 above)
Comment (higs4281) [reply cfgov/ask_cfpb/tests/models/test_pages.py]: facet_map represents the json string that the elasticsearch query returns.
we deserialize that to make the `facets` value in context.

File: cfgov/ask_cfpb/tests/models/test_pages.py (hunk 3 above)
Comment (higs4281) [reply cfgov/ask_cfpb/tests/models/test_pages.py]: good idea!

**Pull request merged**